tests/test-core/utils/js/ConditionalTestUtils.js

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2016 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+
+*/
+
+/* global fluid, jqUnit */
+
+(function () {
+    "use strict";
+
+    fluid.registerNamespace("fluid.test.conditionalTestUtils");
+
+    // "platformArg" is a string to check against navigator.platform
+
+    fluid.test.conditionalTestUtils.isPlatform = function (platformArg) {
+        var detectedPlatform = navigator.platform ? navigator.platform : undefined;
+
+        return detectedPlatform.indexOf(platformArg) >= 0;
+    };
+
+    fluid.test.conditionalTestUtils.isLinux = function () {
+        return fluid.test.conditionalTestUtils.isPlatform("Linux");
+    };
+
+    fluid.test.conditionalTestUtils.isMac = function () {
+        return fluid.test.conditionalTestUtils.isPlatform("Mac");
+    };
+
+    fluid.test.conditionalTestUtils.isWindows = function () {
+        return fluid.test.conditionalTestUtils.isPlatform("Windows");
+    };
+
+    fluid.contextAware.makeChecks({
+        "fluid.platform.isLinux": "fluid.test.conditionalTestUtils.isLinux",
+        "fluid.platform.isMac": "fluid.test.conditionalTestUtils.isMac",
+        "fluid.platform.isWindows": "fluid.test.conditionalTestUtils.isWindows"
+    });
+
+    // Chooses which test function to execute based on the results of a
+    // promise; wraps the promise in an asyncTest to cause QUnit's test
+    // runner to suspend while the decision is being made asynchronously by
+    // the promise. Without this, QUnit will merrily proceed along to the
+    // next test set, which can cause errors various contexts including the
+    // all-tests runner.
+    //
+    // wrapperMessage, task, resolveFunc, rejectFunc: required
+    // "task" must be a function returning a promise
+    // resolveMessage, rejectMessage: optional strings, passed to the test
+    // functions
+
+    fluid.test.conditionalTestUtils.chooseTestByPromiseResult = function (wrapperMessage, task, resolveFunc, rejectFunc, resolveMessage, rejectMessage) {
```

Comment:
I think that this function is sufficiently meaty that we should have a test case for it